api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -755,17 +748,69 @@ public Long getCountOfPatients(String query, boolean includeVoided) {
 		if (StringUtils.isBlank(query)) {
 			return 0L;
 		}
-		
-		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Patient.class);
-		criteria = new PatientSearchCriteria(sessionFactory, criteria).prepareCriteria(query, includeVoided);
-		
-		// Using Hibernate projections did NOT work here, the resulting queries could not be executed due to
-		// missing group-by clauses. Hence the poor man's implementation of counting search results.
-		//
-		return (long) criteria.list().size();
+
+		return (long) findPatients(query, includeVoided).size();
 	}
-	
-	/**
+
+    private List<Patient> findPatients(String query, boolean includeVoided) {
+        query = LuceneQuery.escapeQuery(query);
+
+        LuceneQuery<PatientIdentifier> luceneQuery = getPatientIdentifierLuceneQuery(query, includeVoided);
+		List<PatientIdentifier> patientIdentifiers = luceneQuery.list();
+		LuceneQuery<PersonName> luceneQuery2 = getPersonNameLuceneQuery(query, includeVoided);
+		luceneQuery2.skipSame(luceneQuery.listProjection("patient.personId"), "person.personId");
+		List<PersonName> personNames = luceneQuery2.list();
+		LuceneQuery<PersonAttribute> luceneQuery3 = getPersonAttributeLuceneQuery(query, includeVoided);
+		luceneQuery3.skipSame(luceneQuery.listProjection("patient.personId"), "person.personId");
```

Comment:
I like how it turned out. Is luceneQuery.listProjection cached? If not please do not call it twice.